packages/hdwallet-ledger/src/ledger.ts

```diff
@@ -329,6 +330,38 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
     return res.payload
   }
 
+  /**
+   * Gets name and version of current app open
+   * @returns {Promise<Object>} - payload
+   *                            - payload.name - name of current app open
+   *                            - payload.version - app version
+   */
+  public async getAppAndVersion (): Promise<any> {
+    const res = await this.transport.call(null, 'getAppAndVersion')
+    handleError(res, this.transport)
+    return res.payload
+  }
+
+  /**
+   * Returns Ledger's name for an app given a coin's
+   * ticker symbol
+   * @param symbol - i.e. 'BCH'
+   * @returns appName i.e. "Bitcoin Cash"
+   */
+  public getAppNameBySymbol (symbol: string): string {
+    return appsUtil[symbol]
+  }
+
+  /**
+   * Prompt user to open given app on device
+   * User must be in dashboard
+   * @param appName - human-readable app name i.e. "Bitcoin Cash"
+   */
+  public async openApp (appName: string): Promise<void> {
```

Comment:
Wondering if we should just pass the symbol to this function and let it determine the app name since the lookup table is in here anyways

Reply:
We should also throw an error for symbols we don't have mapped to an app name and toUpperCase the symbol name in case it is passed with different casing

Reply:
IMO these should be the HDWallet name for a coin so the interface across all of HDWallet is uniform. A.k.a names from this list: https://github.com/shapeshift/hdwallet/blob/ba223f94dca0797eef3bb5255cc296a30a0329de/packages/hdwallet-keepkey/src/bitcoin.ts#L44. In `coinMap`, on the shapeshift.com side of things, that is called `keepkeyName`, FWIW. (needs to be renamed `hdwalletName` or something)

Reply:
i.e. this should be `public async openApp(coin: Coin): Promise<void>`

Reply:
I add a validateCurrentApp method